src/ddmd/escape.d

```diff
@@ -299,15 +300,17 @@ bool checkAssignEscape(Scope* sc, Expression e, bool gag)
 
     foreach (VarDeclaration v; er.byref)
     {
-        //printf("byref: %s\n", v.toChars());
+        if (log) printf("byref: %s\n", v.toChars());
         if (v.isDataseg())
             continue;
 
         Dsymbol p = v.toParent2();
 
         // If va's lifetime encloses v's, then error
         if (va &&
-            (va.enclosesLifetimeOf(v) && !(v.storage_class & STCparameter) || va.storage_class & STCref) &&
+            (va.enclosesLifetimeOf(v) && !(v.storage_class & STCparameter) ||
+             va.storage_class & STCref ||
+             va.isDataseg()) &&
```

Comment:
This looks like a special case very much tuned towards the test case. Shouldn't the check for global data be part of enclosesLifetimeOf?